packages/ubuntu_desktop_installer/integration_test/ubuntu_desktop_installer_test.dart

```diff
@@ -0,0 +1,161 @@
+    await tester.selectOption(tester.lang.installUbuntu);
+    await tester.pumpAndSettle();
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+
+    // Keyboard layout
+    expect(find.byType(KeyboardLayoutPage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+
+    // Updates and other software
+    expect(find.byType(UpdatesOtherSoftwarePage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+
+    // Allocate disk space
+    expect(find.byType(AllocateDiskSpacePage), findsOneWidget);
+    await tester.tapHighlightedButton(tester.lang.startInstallingButtonText);
+    await tester.pumpAndSettle();
+
+    // Write changes to disk
+    expect(find.byType(WriteChangesToDiskPage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+
+    // Who are you?
+    expect(find.byType(WhoAreYouPage), findsOneWidget);
+    await tester.enterFieldText(
+        'Real Name', tester.lang.whoAreYouPageRealNameLabel);
+    await tester.enterFieldText(
+        'hostname', tester.lang.whoAreYouPageComputerNameLabel);
+    await tester.enterFieldText(
+        'username', tester.lang.whoAreYouPageUsernameLabel);
+    await tester.enterFieldText(
+        'password', tester.lang.whoAreYouPagePasswordLabel);
+    await tester.enterFieldText(
+        'password', tester.lang.whoAreYouPageConfirmPasswordLabel);
+    await tester.pump();
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+
+    // Choose your look
+    expect(find.byType(ChooseYourLookPage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pump();
+
+    // Installation slides
+    expect(find.byType(InstallationSlidesPage), findsOneWidget);
+    await tester.pumpAndSettle();
+
+    // Installation complete
+    expect(find.byType(InstallationCompletePage), findsOneWidget);
+    await tester.pumpAndSettle();
+
+    final windowClosed = waitForWindowClosed();
+    await tester.tapButton(tester.lang.shutdown);
+    expect(windowClosed, completion(isTrue));
+    await tester.pumpAndSettle();
+  });
+
+  testWidgets('turn off bitlocker', (tester) async {
+    app.main(<String>['--machine-config', 'examples/win10.json']);
+    await waitForSubiquityServer();
+    await tester.pumpAndSettle();
+
+    expect(find.byType(WelcomePage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+    expect(find.byType(TryOrInstallPage), findsOneWidget);
+    await tester.selectOption(tester.lang.installUbuntu);
+    await tester.pumpAndSettle();
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+    expect(find.byType(KeyboardLayoutPage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
+    expect(find.byType(UpdatesOtherSoftwarePage), findsOneWidget);
+    await tester.tapContinue();
+    await tester.pumpAndSettle();
```

Comment:
The sole purpose of this clumsy sequence is to navigate to the _Updates and other software_ page. Perhaps we could add a command-line argument (`--initial-route <route>`) to make it possible to jump straight to a relevant page.

Reply:
https://github.com/canonical/ubuntu-desktop-installer/pull/367